kernel/src/net/socket/unix/stream/connected.rs

```diff
@@ -19,32 +25,49 @@ impl Connected {
     pub(super) fn new_pair(
         addr: Option<UnixSocketAddrBound>,
         peer_addr: Option<UnixSocketAddrBound>,
+        reader_pollee: Option<Pollee>,
+        writer_pollee: Option<Pollee>,
     ) -> (Connected, Connected) {
-        let (writer_this, reader_peer) = Channel::with_capacity(DEFAULT_BUF_SIZE).split();
-        let (writer_peer, reader_this) = Channel::with_capacity(DEFAULT_BUF_SIZE).split();
+        let (writer_peer, reader_this) =
+            Channel::with_capacity_and_pollees(DEFAULT_BUF_SIZE, None, reader_pollee).split();
+        let (writer_this, reader_peer) =
+            Channel::with_capacity_and_pollees(DEFAULT_BUF_SIZE, writer_pollee, None).split();
+
```

Comment:
> The idea here is to enable two Channels to share the same 2 Pollees. Specifically, the `Producer` of the first `Channel` and the `Consumer` of the second `Channel ` would both utilize the same Pollee, and vice versa, right?

Not really. The `reader_pollee` and the `writer_pollee` are different pollees.

This is only because we need the same pollees from the `Init` state to be used in the `Connected` states. Otherwise, if we register a socket with an epoll instance in the `Init` state, we will not be able to receive events after the socket moves to the `Connected` state. That's unfortunate.

Reply:
Thanks for your explanation! Then current implementation makes sense to me!